### PR TITLE
Align the specs site visuals with our Apollo docs look and feel

### PR DESCRIPTION
**Pull request by hwillson**
This PR is a first pass at aligning the look and feel of the specs site with our standard [Apollo docs](https://www.apollographql.com/docs) visuals. Since the specs site is currently generated using a custom process / framework, aligning the sites exactly will be a bit cumbersome, but the changes in this PR should help promote our common Apollo branding a bit further.

Before:

Image 2021-03-31 at 3 22 14 PM

After:

Image 2021-03-31 at 3 22 39 PM

This PR (and the subsequent PR's that will be coming shortly for the `core` and `join` repos) should be considered a stop gap until we have the time to fully convert the spec site over to run on Gatsby, leveraging [`gatsby-theme-apollo`](https://github.com/apollographql/gatsby-theme-apollo).

A few important things to call out:

- This PR attempts to do the minimal needed to get the new visuals in place. It is not an exhaustive refactoring of the existing codebase.
- After some discussion on Slack, I've removed the navigation bar (general consensus was that it isn't needed).
- I've removed parts of the codebase that I believe are no longer needed, and in some cases were causing delays on initial (non-cached) page loads.
- The top left hand Apollo SPECS logo is temporary; Design is tracking down the original source file for this, and I'll replace the custom code I'm using for now with the new logo, when it's ready.
- This repo doesn't have doc previews, so I'm guessing we'll have to do a PR review first, merge things in when we're happy, then do a GH pages publish so everyone can see how it looks? Open to suggestions here for sure.
- I need to do more mobile testing; I'll tack any changes needed on here when I'm done.

### CODE REVIEW

**Comment by hwillson** [comment]
@abernix @glasser @trevor-scheer I'll keep this in draft until the `join` and `core` PR's are submitted, then assign you all for review at the same time. Thanks!

**Comment by hwillson** [comment]
Okay, all PR's should be good to go now.

**Comment by hwillson** [comment]
And here's a quick visual of the landing page:

Image 2021-03-31 at 4 02 05 PM

File: index.html [hunk 1]
```diff
@@ -27,150 +21,192 @@
...
+    </span>
+  </a>
+
+  Core schemas are standard GraphQL schemas which follow the [core specification](core/draft/pre-0). The [core spec](core/draft/pre-0) provides machinery to reference namespaced, versioned directives which provide specified metadata to schema processors and data cores.
+
+  All other specifications in this library are core-compliant specs, designed to be referenced from core schemas.
+
+  ### Joining and Linking
+  <a class=spec href=join/draft/pre-0>
+    <span class=name>join</span>
```
Comment (glasser) [inline index.html:124]: Only core and join exist. You can remove the others. I can maybe work on figuring out how to improve the text above once you've done so. Or I can remove them, whatever works for you.

File: index.html [hunk 2]
```diff
@@ -27,150 +21,192 @@
...
+    </div>
+    <h1 class="title">Technical Specifications</h1>
+  </header>
+
+  ## Specifications
+
+  ### Core Schemas
+  <a class=spec href=core/draft/pre-0>
```
Comment (glasser) [inline index.html:111]: this should link to core/v0.1 (and maybe use quotes? i've never seen HTML that doesn't quote hrefs before...)

(join v0.1 isn't ready yet)

(I can do this too, just want to make sure I remember.)

File: main.js [hunk 3]
```diff
@@ -11,39 +7,11 @@ addEventListener('load', main)
 async function main() {
   removeEventListener('load', main)
 
-  const theme = setTheme('dark')
-
   if (!document.documentElement.dataset.rendered) {
-    const mermaid = import('./mermaid.js')
```
Comment (glasser) [inline main.js:17]: I think mermaid is used in the core spec? It's a graph drawing thing? Does this break it?

**Review by glasser — APPROVED** [review]
I don't think I can usefully review this as a code change with my level of CSS/etc knowledge, but I'm happy to QA the site if you can get it set up somewhere. And honestly if that's hard, you can just merge them all and we can QA them on the live unpublicized site.

**Comment by hwillson** [comment]
Sorry @glasser, I put everything on a test server and mentioned it in Slack, but I'm not sure if you saw it. You can try things out here: http://specs.apollo.ist/

File: index.html (hunk 1 above)
Comment (hwillson) [reply index.html:124]: @glasser would there be any value in leaving them, but disabling them, to show that they will be coming? We could add a "coming soon" descriptor or something.

File: main.js (hunk 3 above)
Comment (hwillson) [reply main.js:17]: @glasser right now mermaid is only be used on the actual spec-md sites themselves, and it's still being loaded there.

File: index.html (hunk 2 above)
Comment (hwillson) [reply index.html:111]: @glasser I found the lack of quotes odd as well, but that's how everything was setup (and it's handled by the custom rendering engine), so I just followed the current code style. Once we get this and the other PR's merged, I'll do a link update pass before publishing.

File: index.html (hunk 1 above)
Comment (glasser) [reply index.html:128]: I don't think there's a plan to implement any of them. They were part of an early proposal by a single team member but haven't been part of the more collaborative iterative process towards finishing.

Comment (hwillson) [reply index.html:128]: Got it, thanks @glasser - I'll remove them.

File: index.html (hunk 2 above)
Comment (hwillson) [reply index.html]: I've updated the core/v0.1 links - thanks!

**Pull request merged**